### PR TITLE
Multi-site LSTM tests using example flight data

### PR DESCRIPTION
**Pull request by lindsayplatt**
My conclusion from all the work shown in this PR is that the best approach that I can find right now to training a multisite LSTM is to have a separate column with the values of interest for each site. I think this will complicate future input features because they will have distinct values that will somehow need to be matched (or maybe the model just does that?).

I've been struggling with figuring out how to run the LSTM with multiple sites and be able to extract predictions unique to each site. I can't seem to find how people do this. I've tried to read through [Frederik Kratzert's code](https://github.com/kratzert/ealstm_regional_modeling/tree/master/papercode) and [Andy McAliley's code](https://github.com/USGS-R/lake-temperature-lstm-static) (which I believe is based on Kratzert's), but haven't been able to figure out exactly how this is done. 

In this PR, I am testing a few different approaches using the flight data example. To do this, I separated them into their own `lstm_modeling/run_*_lstm.py` scripts, so that it would be easy to edit and run each independently. They all used the same number of layers, hidden size, epochs, and learning rate. Here's what I've tried:

- **_Approach 1: Move the value we are trying to predict into its own column for each "site"._** In this approach, I created a column called `airline_american` and `airline_delta`, whose values are the passenger counts. I then build the LSTM with these two columns as features (along with year and month) and then extract them at the end.
- **_Approach 2: Apply a one-hot encoder to the site data._** This approach is similar to the last, except that there is still a `value` column representing the values I would like to predict and the sites are added as their own column (`american` and `delta`) which contain 0s or 1s for which airline each row of data represents. I got this idea from reading [this blog Kaan Kuguoglu](https://towardsdatascience.com/building-rnn-lstm-and-gru-for-time-series-using-pytorch-a46e5b094e7b).
- **_Approach 3: Include site as part of a MultIindex_**: In this approach, the airline type is moved from being a column to being part of the data index (which is where the time component goes to maintain ordering). This is easily done with `df.set_index()` and then reset with `df.reset_index()` in order to make plots.

## Results

From what I have observed here, it seems like the `Approach 1` does a better job of modeling these two datasets. In `Approach 2`, the predictions are not as unique as I would expect them to be between the two datasets which leads me to believe it is not appropriately distinguishing between the sites (they almost look identical which made me think I may have made an error in my plotting code, but if you look closely at the red lines near the beginning of the plot, they are distinct). `Approach 3` has very similar results to `Approach 2` where the predictions between the two airlines are suspiciously close but I know they are distinct (and not a plotting error) because the early values are quite different.

The data for `Delta` that is being fit are all slightly different because they include randomly generated values, but all are fairly similar to this:

image

### Approach 1 in the list above: move the value we are trying to predict into its own column for each "site" and treat as features.

image

image

### Approach 2 in the list above: one-hot encode the site information as feature columns.

image

image

### Approach 3 in the list above: include site as part of a MultIindex

image

image

### CODE REVIEW

**Comment by lindsayplatt** [comment]
What happens when I make the two datasets wildly different? Which of these approaches is better? Note that they all have slightly different data because they are randomly generated values, but all are similar enough to this:

image

### Approach 1: move the value we are trying to predict into its own column for each "site" and treat as features.

image

### Approach 2: one-hot encode the site information as feature columns.

image

### Approach 3: include site as part of a MultIindex

image

**Comment by lindsayplatt** [comment]
Conclusion: having a separate column per site appears to be the most effective way to train a multi-site LSTM. Next, I will see how this does with actual salinity data.
